RIOT-OS/RIOT
# mcufont: Initial addition of MCUFont package

### Contribution description

This adds the MCUFont library as a package to RIOT. Only the decoder part has been packaged. I did not touch the font generation, so the test includes a manually compressed font created via the tooling included in the package.

The included patch files have been submitted as PR upstream in https://github.com/mcufont/mcufont/pull/30

### Testing procedure

Run the test on Native or on any board, The output should be along the lines of:
```
2023-06-11 21:56:40,685 # Help: Press s to start test, r to print it is ready
s
2023-06-11 21:56:42,651 # START
2023-06-11 21:56:42,652 # main(): This is RIOT! (Version: 2023.07-devel-512-g49ee9-pr/mcufont)
2023-06-11 21:56:42,652 # Generated RIOT application: 'mcufont'
2023-06-11 21:56:42,653 #                                                                                                                                 |
2023-06-11 21:56:42,654 #                                                                                                                                 |
2023-06-11 21:56:42,694 #                   █  █                                                                                                          |
2023-06-11 21:56:42,695 #  █     █          █  █                  █████   █    ███   ███████  █                                                           |
2023-06-11 21:56:42,697 #  █     █          █  █                  █    █  █   █   █     █     █                                                           |
2023-06-11 21:56:42,737 #  █     █   ████   █  █   ████           █    █  █  █     █    █     █                                                           |
2023-06-11 21:56:42,739 #  █     █  ██  ██  █  █  ██  ██          █    █  █  █     █    █     █                                                           |
2023-06-11 21:56:42,740 #  ███████  █    █  █  █  █    █          █████   █  █     █    █     █                                                           |
2023-06-11 21:56:42,781 #  █     █  ██████  █  █  █    █          █   █   █  █     █    █     █                                                           |
2023-06-11 21:56:42,782 #  █     █  █       █  █  █    █          █    █  █  █     █    █                                                                 |
2023-06-11 21:56:42,783 #  █     █  ██   █  █  █  ██  ██  █       █    █  █   █   █     █     █                                                           |
2023-06-11 21:56:42,807 #  █     █   ████   █  █   ████   █       █     █ █    ███      █     █                                                           |
2023-06-11 21:56:42,807 # { "threads": [{ "name": "idle", "stack_size": 256, "stack_used": 220 }]}
2023-06-11 21:56:42,808 # { "threads": [{ "name": "main", "stack_size": 1280, "stack_used": 436 }]}
```


### Issues/PRs references

None
